tests/installation/test_executor.py

```diff
@@ -701,12 +701,27 @@ def test_executor_should_write_pep610_url_references_for_urls(
         source_type="url",
         source_url="https://files.pythonhosted.org/demo-0.1.0-py2.py3-none-any.whl",
     )
+    # Set package.files so the executor will attempt to hash the package
```

Comment:
Can you extend `test_executor_should_write_pep610_url_references_for_files` in a similar way, please?

Reply:
For this to work, I needed to populate the `_hashes` dictionary inside function `_prepare_archive`. Hope that's fine.

Reply:
LGTM. Apparently, we did write the hash only for URL dependencies but not for file dependencies so far because of the missing part in `_prepare_archive`.